src/utils/event-registrar.js

```diff
@@ -81,15 +81,21 @@ export default class EventRegistrar {
         event.handled = true;
         immediatePropagationStopped = true;
       };
+
       for (let i = 0; i < entries.length; i++) {
-        const {type, handler} = entries[i];
+        const {type, handler, once} = entries[i];
         handler(
           Object.assign({}, event, {
             type,
             stopPropagation,
             stopImmediatePropagation
           })
         );
+        if (once) {
+          this.remove(type, handler);
+          // remove() mutates the entries array by deleting the current entry
+          i--;
```

Comment:
I think this `i--` works fine here, but I typically iterate backwards through an array when I may be removing an element, to avoid side effects from the removal that unexpectedly change the order / current index.

E.g. if there's an error in `this.remove()` and the element is not removed, you have an infinite loop.